setup.ts

```diff
@@ -0,0 +1,68 @@
+import { writeFileSync } from 'fs'
+import minimist from 'minimist'
+import { createInterface } from 'readline'
+import { Writable } from 'stream'
+
+import { Config, configSample } from './src/config.schema'
+
+const argv = minimist(process.argv.slice(2))
+
+const isProduction = argv.dev !== true
+const password: string | boolean = argv.p ?? argv.password
```

Comment:
dont allow password to be entered on the command line as this would get logged to paper trail

Reply:
Better to get the password from an environment variable

Reply:
If we look at how `yarn setup` is being used at the call-site, you'll notice we're already using an env var at the call-site: https://github.com/EdgeApp/edge-devops/pull/2/files#diff-6be1069dd5f798d373a0b894a0d1ed0e7efcd92e5f256306fd8e63c13b35e3dfR14 (hopefully that link takes you to line 14 of sync-server/install-sync-server.sh). In that case, we're already are protected by how we this sensitive data propagates the installation/setup of a sync-server by virtue that our sensitive data is strictly being initialized as an env var at the highest-level shell script run on the server by passing the script as a user_data DigitalOcean parameter: https://github.com/EdgeApp/edge-devops/pull/2/files#diff-9fc15ba954cca7a22fd4c32c61b15c9936581bdc2dcce35a5c1b64194597033fR173 (line 173-179 of provisions/sync-server-digitalocean.ts). Our env vars are set before any software installation or potential logging setup, so it should be kept secret/safe from exposure.

If we change our `yarn setup` to only accept an env var, we still must guarantee that the env var is set before any logging setup (which we are doing already). Because the arg is set to an env var, it inherits the security of the env var, so no change is needed to improve any measure of security. Making the change from arg to env var doesn't really have any effect other than preference.

Reply:
You're right. It should be a env var by convention to make it easier to add more variables to the setup if needed.